fabric-networking-api-v1/src/main/java/net/fabricmc/fabric/mixin/networking/ServerCommonNetworkHandlerMixin.java

```diff
@@ -57,4 +76,24 @@ private void onPlayPong(CommonPongC2SPacket packet, CallbackInfo ci) {
 			addon.onPong(packet.getParameter());
 		}
 	}
+
+	@Inject(method = "onCookieResponse", at = @At("HEAD"))
```

Comment:
You should make this cancelable, and pass the `CookieResponseC2SPacket` to the addon.

If the addon is able to handle it cancel the method, if not allow vanilla to disconnect.